packages/teleport-test/src/standalone.ts

```diff
@@ -91,6 +92,16 @@ const run = async () => {
           ...packerOptions.publishOptions,
           projectSlug: 'teleport-project-next-cms',
         },
+        plugins: [
+          new ProjectPluginCustomFiles([
+            {
+              id: 'teleport-config.json',
+              name: 'teleport-config.json',
+              path: [],
+              content: JSON.stringify({ '/about': { path: '/about' } }, null, 2),
```

Comment:
Why do we need this? I'm just trying to understand what happens under the hood...

Reply:
So, the mapping between `contentType` from the CMS and the `route` in the projects are mutable. Means, users can change them any time.
Eg: contentType of `BlogPost` can become `/some-random-route`. 

This info, is only known to our backend and not contentful. So, contentful webhook is always going to call with `BlogPost`. In order to figure these mappings out. GUI is going to inject a file called `teleport-config.json`. Which is picked up by the `webhook` handler and then maps the contentTypes to their respective routes. More about the implementation is here
https://github.com/teleporthq/teleport-cms-mappers/pull/3/files#diff-8c76daa4bb77de60c6b376dfdbbf7faec4446fbcce1c5baf9169dcc7aae32561

Reply:
This particular change here is just to test the feature out. But in general this is exactly the place where GUI injects it's file. As a fallback just in case if GUI fails to inejct but the `revalidate` is enabled. We inject a empty JSON file
https://github.com/teleporthq/teleport-code-generators/pull/817/files#diff-f81aa656e26fb0ab234fcf65a5c0624f75b35a9430b1e32684f570f91a955b87R43

Reason, we use static imports
```js
import routeMapper from '../../teleport-config.json'
```

Which breaks the whole build because of NextJS static analysis at build time if the file is missing. We can't use `readFileSync` because the file is not added to the `.next` at the end of build process. Webpack includes a file only if is `imported` so. We need to always ensure there is some sort of file if not the injected file to make the builds pass.

Probably not need the whole explanation 😅, but just wrote it anyways. As we can use as future reference 😀